client/js/components/statement/listStatements/ListStatements.vue

```diff
@@ -486,6 +487,54 @@ export default {
       }
     },
 
+    /**
+     * If statement already has segments, do nothing
+     * If statement is not claimed, silently assign it to the current user and go to split statement view
+     * If statement is claimed by current user, go to split statement view
+     * If statement is claimed by another user, ask the current user to claim it and, after they confirm, assign it to
+     * the current user, then go to split statement view. If the user doesn't confirm, do nothing.
+     * @param statementId {string}
+     * @param assignee {object} { data: { id: string, type: string }}
+     * @param segmentsCount {number || string} can be a number or '-'
+     */
+    handleStatementSegmentation (statementId, assignee, segmentsCount) {
+      const isStatementSegmented = segmentsCount > 0 && segmentsCount !== '-'
+      const isStatementClaimedByCurrentUser = assignee.id === this.currentUserId
+      const isStatementClaimed = assignee.id !== ''
+      const isStatementClaimedByOtherUser = isStatementClaimed && !isStatementClaimedByCurrentUser
+
+      if (isStatementSegmented) {
+        return
+      }
+
+      if (isStatementClaimedByCurrentUser) {
+        window.location.href = Routing.generate('dplan_drafts_list_edit', { statementId: statementId, procedureId: this.procedureId })
+      }
+
+      if (!isStatementClaimed) {
+        // claim silently
+        this.claimStatement(statementId)
+          .then(() => {
+            window.location.href = Routing.generate('dplan_drafts_list_edit', { statementId: statementId, procedureId: this.procedureId })
+          })
+          .catch(err => {
+            console.error(err)
+          })
```

Comment:
Can You extract that, so its not  doubled with the block below?

Reply:
Or maybe You can merge the two if-statements since the content looks the same

Reply:
good point. done in 5a9ae6fb3afc45e984f5aa24bb5a7c2f5f036e10